modules/core/src/lifecycle/create-props.ts

```diff
@@ -46,22 +57,40 @@ export function createProps<T>(
   return propsInstance;
 }
 
+const MergedDefaultPropsCacheKey = '_mergedDefaultProps';
+
 // Return precalculated defaultProps and propType objects if available
 // build them if needed
-function getPropsPrototype(componentClass) {
-  const defaultProps = getOwnProperty(componentClass, '_mergedDefaultProps');
+function getPropsPrototype(componentClass, extensions?: any[]) {
+  // A string that uniquely identifies the extensions involved
+  let cacheKey = MergedDefaultPropsCacheKey;
+  if (extensions) {
+    for (const extension of extensions) {
```

Comment:
Sort the extensions so that `_mergedDefaultProps:ExtensionA:ExtensionB` doesn't duplicate `_mergedDefaultProps:ExtensionB:ExtensionA`?

Reply:
- Sort is going to have a much bigger perf impact. The key generation could be invoked indefinitely and every frequently, while caching only happen once or twice (due to duplicates). It is also unlikely that the user keep changing the order of extensions upon rerender.
- Technically speaking the order does matter, if the extensions define overlapping props.

Reply:
Good point regarding overlapping props